### PR TITLE
Add inline example of PyTorch pyfunc usage

### PR DESCRIPTION
**Pull request by Gonmeso**
Signed-off-by: Gonzalo Mellizo-Soto <gmellizosoto@gmail.com>

## Related Issues/PRs

Contributes to #6099 

## What changes are proposed in this pull request?

This PR adds an inline usage example to the PyTorch model flavour.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

Adds usage example to the PyTorch model flavour

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -703,6 +703,52 @@ via :py:func:`mlflow.pyfunc.load_model()`.
...
+        loss = loss_function(outputs, y)
+        loss.backward()
+
+        optimizer.step()
+
+    with mlflow.start_run() as run:
+        mlflow.pytorch.save_model(net, "net")
+
+    pytorch_pyfunc = mlflow.pyfunc.load_model(model_uri="net")
+
```
Comment (harupy) [inline docs/source/models.rst:749]:
```suggestion
    with mlflow.start_run() as run:
        model_info = mlflow.pytorch.log_model(net, "model")

    pytorch_pyfunc = mlflow.pyfunc.load_model(model_uri=model_info.model_uri)

```

File: docs/source/models.rst [hunk 2]
```diff
@@ -703,6 +703,52 @@ via :py:func:`mlflow.pyfunc.load_model()`.
...
+    predictions = pytorch_pyfunc.predict(np.array([3.4, -1.2, 4.8, 7.1, 9.4, -6.9], np.float32))
```
Comment (harupy) [inline docs/source/models.rst:750]:
```suggestion
    predictions = pytorch_pyfunc.predict(np.array([3.4, -1.2, 4.8, 7.1, 9.4, -6.9], np.float32))
    print(predictions)
```

File: docs/source/models.rst [hunk 3]
```diff
@@ -703,6 +703,52 @@ via :py:func:`mlflow.pyfunc.load_model()`.
...
+    class Net(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.l1 = nn.Linear(6, 1)
+
+        def forward(self, x):
+            return self.l1(x)
+
+
+    net = Net()
```
Comment (harupy) [inline docs/source/models.rst:728]:
```suggestion
    net = nn.Linear(6, 1)
```

Does this work?

File: docs/source/models.rst [hunk 4]
```diff
@@ -703,6 +703,52 @@ via :py:func:`mlflow.pyfunc.load_model()`.
     In case of multi gpu training, ensure to save the model only with global rank 0 gpu. This avoids
     logging multiple copies of the same model.
 
+PyTorch pyfunc usage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (harupy) [inline docs/source/models.rst:707]:
```suggestion
PyTorch pyfunc usage
~~~~~~~~~~~~~~~~~~~~
```

**Review by harupy — APPROVED** [review]

**Comment by harupy** [comment]
LGTM!

**Comment by Gonmeso** [comment]
Hi @harupy ,

I ran into some problems and couldn't review your suggestions, sorry for the delay and thank you for keeping this moving forward.

**Pull request merged**